go-controller/pkg/kube/kube.go

```diff
@@ -318,17 +339,41 @@ func (k *Kube) GetEgressIP(name string) (*egressipv1.EgressIP, error) {
 }
 
 // GetEgressIPs returns the list of all EgressIP objects from kubernetes
-func (k *Kube) GetEgressIPs() (*egressipv1.EgressIPList, error) {
-	return k.EIPClient.K8sV1().EgressIPs().List(context.TODO(), metav1.ListOptions{
+func (k *Kube) GetEgressIPs() ([]*egressipv1.EgressIP, error) {
+	list := []*egressipv1.EgressIP{}
+	err := pager.New(func(ctx context.Context, opts metav1.ListOptions) (runtime.Object, error) {
+		return k.EIPClient.K8sV1().EgressIPs().List(ctx, opts)
+	}).EachListItem(context.TODO(), metav1.ListOptions{
 		ResourceVersion: "0",
+	}, func(obj runtime.Object) error {
+		list = append(list, obj.(*egressipv1.EgressIP))
+		return nil
 	})
+	return list, err
 }
 
 // GetEgressFirewalls returns the list of all EgressFirewall objects from kubernetes
-func (k *Kube) GetEgressFirewalls() (*egressfirewall.EgressFirewallList, error) {
-	return k.EgressFirewallClient.K8sV1().EgressFirewalls(metav1.NamespaceAll).List(context.TODO(), metav1.ListOptions{
+func (k *Kube) GetEgressFirewalls() ([]*egressfirewall.EgressFirewall, error) {
+	list := []*egressfirewall.EgressFirewall{}
+	err := pager.New(func(ctx context.Context, opts metav1.ListOptions) (runtime.Object, error) {
+		return k.EgressFirewallClient.K8sV1().EgressFirewalls(metav1.NamespaceAll).List(ctx, opts)
+	}).EachListItem(context.TODO(), metav1.ListOptions{
 		ResourceVersion: "0",
+	}, func(obj runtime.Object) error {
+		list = append(list, obj.(*egressfirewall.EgressFirewall))
+		return nil
 	})
+	return list, err
+}
+
+// GetEndpoint returns the Endpoints resource
+func (k *Kube) GetEndpoint(namespace, name string) (*kapi.Endpoints, error) {
+	return k.KClient.CoreV1().Endpoints(namespace).Get(context.TODO(), name, metav1.GetOptions{})
+}
+
+// CreateEndpoint creates the Endpoints resource
+func (k *Kube) CreateEndpoint(namespace string, ep *kapi.Endpoints) (*kapi.Endpoints, error) {
+	return k.KClient.CoreV1().Endpoints(namespace).Create(context.TODO(), ep, metav1.CreateOptions{})
 }
```

Comment:
nit: these were added but I don't see them being used or in the interface? Are they there to implement some other interface?